components/engine_rocks/src/raft_engine.rs

```diff
@@ -233,6 +257,13 @@ impl RaftLogBatch for RocksWriteBatch {
     }
 
     fn cut_logs(&mut self, raft_group_id: u64, from: u64, to: u64) {
+        {
+            let mut ctx = RAFT_LOG_GC_CONTEXT.write().unwrap();
+            if ctx.gc_on_compaction {
```

Comment:
Seems to me we can avoid a lock when accessing read-only `gc_on_compaction`.

Reply:
`gc_on_compaction` is a mutable static value, although it will not be changed after parsing the config, a lock is needed to make the compiler happy.

What about make `gc_on_compaction` an AtomicBool to avoid locking?